tools/bin/ecflow/processes/main/post_processing/tm_analysis.py

```diff
@@ -0,0 +1,139 @@
+direc = config_dict['VIC']['OutputDirRoot']
+n_days = int(config_dict['ECFLOW']['Met_Delay'])
+
+# number of plotting positions
+num_pp = int(config_dict['PERCENTILES']['num_plot_pos'])
+cdf_loc = config_dict['PERCENTILES']['cdf_Total_Moist']
+outfile_loc = config_dict['PERCENTILES']['Percentile_Loc']
+
+vic_start_date = config_dict['VIC']['vic_start_date']
+
+# how many days behind metdata is from realtime
+date_unformat = datetime.now() - timedelta(days=n_days)
+
+date = date_unformat.strftime('%Y-%m-%d')
+month_day = date_unformat.strftime("%B_%-d")
+
+# read VIC output
+f = "fluxes.%s.nc" % (vic_start_date)
+ds = xr.open_dataset(os.path.join(direc, f))
+
+# create list of all latitudes and longitudes for full rectangle
+un_lat = ds['lat'].values
+un_lon = ds['lon'].values
+
+num_lat = len(un_lat)
+num_lon = len(un_lon)
+
+latitude = np.repeat(un_lat, num_lon)
+longitude = np.tile(un_lon, num_lat)
+
+# Weibull Plotting Position
+q = np.arange(1,num_pp+1)/(num_pp + 1.0)
+
+# create a min and max plotting position
+# for any values that fall outside of historic range
+min_q = min(q) / 2
+max_q = max(q) + min_q
+
+# select out time and variable data
+ds_day = ds.sel(time=date)
+tm_ds = ds_day['OUT_SOIL_MOIST'].sum(dim='nlayer') + ds_day['OUT_SWE']
+
+# create a dictionary containing the lat, lon and corresponding percentile
+# value 
+d = []
+
+for lat, lon in zip(latitude, longitude):
+
+    # iterate through all latitudes and longitudes
+    ds_lat = tm_ds.sel(lat=lat)
+    ds_lon = ds_lat.sel(lon=lon)
+
+    value = ds_lon.values
+
+    # read in sorted list of cdf values
+    # if cdf cannot be read then that lat lon is saved in the dictionary without a percentile
+    # this is done to make creating the xarray dataset easier later on
+    try:
+        cdf_file = "%s_%s" % (lat, lon)
+        cdf_path = os.path.join(cdf_loc, month_day, cdf_file)
+        cdf = pd.read_csv(cdf_path, index_col=None,
+                          delimiter=None, header=None)
+        x = cdf[0]
+
+        try:
+            # interpolate percentile based on where value falls
+            # relative to historic range
+            f = interp1d(x, q)
+            percentile = f(value)
+            combine = (lat, lon, float(percentile))
+            d.append(combine)
+
+            # if interpolation fails then a value is assigned based on
+            # whether it is higher or lower than the range
+        except ValueError:
+            if (value > max(x)):
+                percentile = (max_q)
+                combine = (lat, lon, percentile)
+                d.append(combine)
+            else:
+                percentile = (min_q)
+                combine = (lat, lon, percentile)
+                d.append(combine)
+    except IOError:
+        percentile = value
+        combine = (lat, lon)
+        d.append(combine)
+
+# Dictionary to DataFrame to Dataset
+df = pd.DataFrame(d, columns=["Latitude", "Longitude", "tmpercentile"])
```

Comment:
Good idea. I need to make changes to this routine in my next pull request, will make sure to include a function for it.